packages/@okta/vuepress-site/docs/guides/secure-oauth-between-orgs/main/index.md

```diff
@@ -0,0 +1,322 @@
+
+`POST /api/v1/apps/${yourOrg2OrgAppId}/credentials/keys/generate?validityYears=${validYears}`
+
+##### Response example
+
+```json
+{
+    "kty": "RSA",
+    "created": "2022-01-20T19:50:14.000Z",
+    "lastUpdated": "2022-01-20T19:50:14.000Z",
+    "expiresAt": "2024-01-20T19:50:13.000Z",
+    "kid": "sf-jWwRKMUU5588aokhj-xu_mGucHLxIh_-fYLAofB8",
+    "use": "sig",
+    "x5c": [
+        "MIIDqDCCApCgAwIBAgIGAX55C ... Iuo9j3wpemDSgGapXQ=="
+    ],
+    "x5t#S256": "v-v2V8soFmXuhCAhta8wycA9lXKnrJ4ho-N3P8aASFc",
+    "e": "AQAB",
+    "n": "gIxwqCNkdAb1ioyNBY2boqUCrMj_NS ... FiAYF7p_k3XMXOh-hsL_D8FDQ"
+}
+```
+
+> **Note**: The keys are truncated for brevity. Save the `kty`, `e`, `use`, `n`, and `kid` properties to update the keys in your Org2Org and service apps.
+
+From the response of your POST request, copy the `kid` property and [activate the new key by updating the Org2Org app](/docs/reference/api/apps/#update-key-credential-for-application).
+
+##### Request example
+
+```bash
+curl -X PUT \
+  -H 'Accept: application/json' \
+  -H "Authorization: SSWS ${spokeApiToken}" \
+  -H 'Content-Type: application/json' \
+  -d ' {
+    "name": "okta_org2org",
+    "label": "'${spokeOrg2OrgClientLabel}'",
+    "credentials": {
+        "signing": {
+            "kid": "'${kid}'"
+        }
+    }
+}' "https://${yourSpokeOrgDomain}/api/v1/apps/${yourOrg2OrgAppId}"
+```
+
+> **Note**: You must specify `name` and `label` body parameters when you update an Org2Org app.
+
+Then go to your hub org and register the new key to your corresponding service app by using the [Dynamic Client Registration API](/docs/reference/api/oauth-clients/).
+
+##### Request example
+
+```bash
+curl -X PUT \
+  -H 'Accept: application/json' \
+  -H "Authorization: SSWS ${hubApiToken}" \
+  -H 'Content-Type: application/json' \
+  -d ' {
+    "client_id": "'${yourServiceAppId}'",
+    "client_name": "'${hubServiceClientLabel}'",
+    "response_types": [
+      "token"
+    ],
+    "grant_types": [
+      "client_credentials"
+    ],
+    "token_endpoint_auth_method": "private_key_jwt",
+    "application_type": "service",
+    "jwks": {
+                "keys": [
+                  {
+                    "kty": "'${kty}'",
+                    "kid": "'${kid}'",
+                    "e": "'${e}'",
+                    "alg": "RS256",
+                    "use": "'${use}'",
+                    "n": "'${n}'"
+                  }
+                ]
+            }
+ }' "https://${yourHubOrgDomain}/oauth2/v1/clients/${yourServiceAppId}"
+```
+
+> **Note**: When you update the keys in the service app `jwks` property, all the old keys are overwritten. You must specify all settings when you update a client app. Partial updates aren't supported. If any settings are missing when you update a client app, the update fails.
```

Comment:
For Key Rotation, if they follow these steps
1. generate new key
2. update Org2Org's kid
3. register new key with service app

there will be some downtime between step 2 and 3
can we swap step 2 and 3:
1. generate new key
2. register new key with service app
3. update Org2Org's kid 

and add a note: If you want to achieve zero downtime for key rotation, you can update service app with both old and new key in step2(`jwks.keys` is an array, different keys are identified by `kid`), then: 
4. (Optional) update service app to remove the old key

Reply:
I added these notes in the "Key Rotation" section. Thanks for the feedback!